R/utils.R

```diff
@@ -395,57 +395,31 @@ guess_geo_type <- function(geo_value) {
     }
   }
 
-  # If we got here then we failed
   return("custom")
 }
 
-guess_time_type <- function(time_value) {
-  # Convert character time values to Date or POSIXct
-  if (is.character(time_value)) {
-    if (nchar(time_value[1]) <= 10L) {
-      new_time_value <- tryCatch(
-        {
-          as.Date(time_value)
-        },
-        error = function(e) NULL
-      )
-    } else {
-      new_time_value <- tryCatch(
-        {
-          as.POSIXct(time_value)
-        },
-        error = function(e) NULL
-      )
-    }
-    if (!is.null(new_time_value)) time_value <- new_time_value
-  }
 
-  # Now, if a POSIXct class, then use "day-time"
-  if (inherits(time_value, "POSIXct")) {
-    return("day-time")
-  } else if (inherits(time_value, "Date")) {
-    # Else, if a Date class, then use "week" or "day" depending on gaps
-    # Convert to numeric so we can use the modulo operator.
+guess_time_type <- function(time_value, time_value_arg = rlang::caller_arg(time_value)) {
+  if (inherits(time_value, "Date")) {
+    # Else, if a Date class, then use "week" or "day" depending on gaps.
     unique_time_gaps <- as.numeric(diff(sort(unique(time_value))))
-    # We need to check the modulus of `unique_time_gaps` in case there are
-    # missing dates. Gaps in a weekly date sequence will cause some diffs to
-    # be larger than 7 days. If we just check if `diffs == 7`, it will fail
-    # unless the weekly date sequence is already complete.
+    # Gaps in a weekly date sequence will cause some diffs to be larger than 7
+    # days, so check modulo 7 equality, rather than equality with 7.
     return(ifelse(all(unique_time_gaps %% 7 == 0), "week", "day"))
```

Comment:
I'm thinking the logic here will be: if all of the gaps are 0 mod 7, then assume week, otherwise assume day, but give a warning if less than half of the unique time gaps are 1.